core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java

```diff
@@ -510,11 +510,19 @@ public List<QualifiedObjectName> listTables(Session session, QualifiedTablePrefi
 
         Optional<QualifiedObjectName> objectName = prefix.asQualifiedObjectName();
         if (objectName.isPresent()) {
-            Optional<RelationType> relationType = getRelationTypeIfExists(session, objectName.get());
-            if (relationType.isPresent()) {
-                return ImmutableList.of(objectName.get());
+            try {
+                Optional<RelationType> relationType = getRelationTypeIfExists(session, objectName.get());
+                if (relationType.isPresent()) {
+                    return ImmutableList.of(objectName.get());
+                }
+                // TODO we can probably return empty list here
+            }
+            catch (RuntimeException e) {
+                handleListingError(e, prefix);
+                // TODO This should be improved to not return empty results https://github.com/trinodb/trino/issues/21990
```

Comment:
( https://github.com/trinodb/trino/issues/21990 )

consider linking to https://github.com/trinodb/trino/issues/6551 instead

but i will actually do not add TODO here. I don't know what we should do instead of what we're doing now